[9.0][FIX]stock_auto_move: fix action_done for automatique moves with lots

zakiuu opened the pull request:
The **current behavior**: the `stock_auto_move` module allows to tag a movement to be processed automatically. In a situation of chained movements (ex: reception in two steps) with product tracking set to `Serial` or `Lot`, the automatic movements can't be processed, and a message error is raised to the user: `You have a difference between the quantity on the operation and the quantities specified for the lots.`
Technically speaking **this issue comes from** the fact there is no quantity set on the `pack_lot_ids` of the operations linked to the movement `linked_move_operation_ids`.

**In this PR** I've tried to **fix this issue** by providing a method that auto fills the quantity on the `pack_lot_ids`, and also fill the `qty_done` of the operations using a new method that I've added called `_auto_fill_pack_lot_ids_qty` defined on the `stock.pack.operation` model. This method is called from the `action_done` of the `stock.move` model.

npiganeau committed:
Added stock_auto_move module

Commit:
stock_auto_move : OCA compliant

Commit:
stock_auto_move : replace utf8 by utf-8, use camelcase in class name

npiganeau committed:
Refactorings

Commit:
stock_auto_move: externalize change group to be able to overload it

npiganeau committed:
Use new api mapped instead of ids list comprehension

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove en.po that was erroneously created by transbot

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

zakmak88 committed:
[MIG] stock_auto_move to v9.0

zakmak88 committed:
[IMP]stock_auto_move: auto move from push rules

zakmak88 committed:
[MIG]silog_auto_move: make module installable

zakmak88 committed:
[FIX]stock_auto_move: fix stock_move action_assing method

zakmak88 committed:
[FIX]stock_auto_move: assign automatic procurement to only auto moves without proc.

zakmak88 committed:
[FIX]stock_auto_move: fix action_done for automatique moves with lots

zakiuu commented:
this PR is based on #359

lmignon commented on stock_auto_move/models/stock_move.py:
```diff
@@ -24,6 +24,21 @@ def action_assign(self, no_prepare=False):
+            lambda m: m.state == 'assigned' and m.auto_move and
+            m.product_id.tracking != 'none')
+        auto_moves_operations = \
+            auto_moves.mapped('linked_move_operation_ids').\
```
@zakiuu `auto_moves.mapped('linked_move_operation_ids.operation_id')`

lmignon commented:
@zakiuu You changes makes sens. Can you check to the Travis error

zakmak88 committed:
[REF]stock_auto_move: remove override of _apply method

* The `_apply` method should not be overridden. In case of three step
operation with the last step set to auto move. The override of the
`_apply` method was causing the second move to be automatic as well.

zakiuu commented:
closed by #373
